front/src/app/components/forms/agency/AddAgencyForm.tsx

```diff
@@ -148,19 +152,26 @@ const AgencyForm = ({
 
   const { handleSubmit, formState, reset } = methods;
 
-  const agenciesRetrieverMemoized = useCallback(
-    (departmentCode: DepartmentCode) =>
-      outOfReduxDependencies.agencyGateway.getFilteredAgencies({
-        departmentCode,
-        kind: "withoutRefersToAgency",
-      }),
-    [],
-  );
+  const dispatch = useDispatch();
+  const agencyOptions = useAppSelector(agenciesSelectors.options);
+  const isLoading = useAppSelector(agenciesSelectors.isLoading);
+  const feedback = useAppSelector(agenciesSelectors.feedback);
 
   useEffect(() => {
     reset(formInitialValues);
   }, [reset, formInitialValues]);
 
+  const onDepartmentCodeChangedMemoized = useMemo(
+    () => (departmentCode: DepartmentCode) =>
+      dispatch(
+        agenciesSlice.actions.fetchAgencyOptionsRequested({
+          departmentCode,
+          kind: "withoutRefersToAgency",
+        }),
+      ),
+    [dispatch],
+  );
```

Comment:
tu peux gagner un peu en lisibilité avec `useCallback` :
```suggestion
  const onDepartmentCodeChangedMemoized = useCallback(
    (departmentCode: DepartmentCode) =>
      dispatch(
        agenciesSlice.actions.fetchAgencyOptionsRequested({
          departmentCode,
          kind: "withoutRefersToAgency",
        }),
      ),
    [dispatch],
  );
```